From patchwork Tue Aug  4 12:44:15 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 11700369
Return-Path: <SRS0=wIAV=BO=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 539A9722
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:46:36 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 4C722208C7
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:46:36 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=linaro.org header.i=@linaro.org
 header.b="T8YDxwS+"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726210AbgHDMqQ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Aug 2020 08:46:16 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:41192 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1728311AbgHDMo3 (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Aug 2020 08:44:29 -0400
Received: from mail-wm1-x342.google.com (mail-wm1-x342.google.com
 [IPv6:2a00:1450:4864:20::342])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 0B82AC0617A0
        for <kvm@vger.kernel.org>; Tue,  4 Aug 2020 05:44:24 -0700 (PDT)
Received: by mail-wm1-x342.google.com with SMTP id x5so2548582wmi.2
        for <kvm@vger.kernel.org>; Tue, 04 Aug 2020 05:44:23 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=bN3d8Ia77BH2nBiHDQ6zBNPByWQO01hAGf1Nc3k/FjU=;
        b=T8YDxwS+GAi/l87EqEDVnHA6L2vWdoJLRomogp4zswguktwRhtfY6a3S80oC+la+7Y
         uBOGZPkWUazcUyhEPQsfO3QZ16muTbOfxNxa4uXdYwfXdmkfVzoMRSW4m/T7f0i4RLf7
         dX+k8C7kUSi09yoGq/WkXRUuQWQIUOu80qSJ8B9K9QLhl5toYoFUrNUqUcTwdRw8GrmH
         fMUA0SnmjEveRBv57i02a+TxYyxNXCrEDBzYs7JuFuOONWrjDdCFY0ND5r47wpeLc24V
         fJMl0bur8BNgusZj++pQeoEnXBDmf1q7b4PbFj3Wzq0ri29fof8CCFp8ai4icK9pDdY1
         JhtQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=bN3d8Ia77BH2nBiHDQ6zBNPByWQO01hAGf1Nc3k/FjU=;
        b=NkaEweb9yetJcUncRn7xsPJo7ymD3GUjbl9IZe+2HEpVuVFV+IE6UcvnyaXluCL3Tx
         g6zXtw1z/iuMXszvN7OD7dRELw5wee+EEal0x8odhMV0HhLDQsmQzTONs71Sh81tFGbY
         dBGyh+hTBFr0HvIKirwoQdAJ6ebGGhSrmoyVfnFRLldz0vFIfcCkMeSqDNMVn2Ohc4z7
         7LXzQgKlC73DwJ96ZtdzFCCBTomrM4iU9YPdVjBxrvw0kBtb9ehc/v4sXKk/uU/mRmHc
         L8kdK6hZoAm0lta5pzetmygQcISg04wq7RoAr8t3ktQ2wj+aCbq+0DeaNAgWM0pTTHLs
         YZnQ==
X-Gm-Message-State: AOAM532/u1CQkhlqOb3Obxr5KFWmE7p8WuffHQxzYGATSZjoephTTEYS
        06XLg+jjBTF/ziO2hWYs8wLudg==
X-Google-Smtp-Source: 
 ABdhPJz9A0fUr0s+57vaSQaPT8GQEVwMIBWtT6cib6RQGYxAGZypLJ4R2NHV8/yEEKGRo0R1n0bqCQ==
X-Received: by 2002:a7b:c257:: with SMTP id b23mr3829607wmj.164.1596545062751;
        Tue, 04 Aug 2020 05:44:22 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 b137sm4843577wmb.9.2020.08.04.05.44.18
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 04 Aug 2020 05:44:18 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 860581FF87;
        Tue,  4 Aug 2020 13:44:17 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org, linux-arm-kernel@lists.infradead.org
Cc: linux-kernel@vger.kernel.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>,
 Robert Richter <rrichter@marvell.com>, linux-pci@vger.kernel.org
Subject: [PATCH  v1 1/3] arm64: allow de-selection of ThunderX PCI controllers
Date: Tue,  4 Aug 2020 13:44:15 +0100
Message-Id: <20200804124417.27102-2-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20200804124417.27102-1-alex.bennee@linaro.org>
References: <20200804124417.27102-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

For a pure VirtIO guest bringing in all the PCI quirk handling adds a
significant amount of bloat to kernel we don't need. Solve this by
adding a CONFIG symbol for the ThunderX PCI devices and allowing it to
be turned off. Saving over 300k from the uncompressed vmlinux:

  -rwxr-xr-x 1 alex alex  85652472 Aug  3 16:48 vmlinux*
  -rwxr-xr-x 1 alex alex  86033880 Aug  3 16:39 vmlinux.orig*

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
Cc: Robert Richter <rrichter@marvell.com>
Cc: linux-pci@vger.kernel.org
Cc: linux-arm-kernel@lists.infradead.org
---
 arch/arm64/Kconfig.platforms    | 2 ++
 arch/arm64/configs/defconfig    | 1 +
 drivers/pci/controller/Kconfig  | 7 +++++++
 drivers/pci/controller/Makefile | 4 ++--
 4 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/Kconfig.platforms b/arch/arm64/Kconfig.platforms
index 8dd05b2a925c..a328eebdaa59 100644
--- a/arch/arm64/Kconfig.platforms
+++ b/arch/arm64/Kconfig.platforms
@@ -253,12 +253,14 @@ config ARCH_SPRD
 
 config ARCH_THUNDER
 	bool "Cavium Inc. Thunder SoC Family"
+        select PCI_THUNDER
 	help
 	  This enables support for Cavium's Thunder Family of SoCs.
 
 config ARCH_THUNDER2
 	bool "Cavium ThunderX2 Server Processors"
 	select GPIOLIB
+        select PCI_THUNDER
 	help
 	  This enables support for Cavium's ThunderX2 CN99XX family of
 	  server processors.
diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index 2ca7ba69c318..d840cba99941 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -199,6 +199,7 @@ CONFIG_PCI_HOST_GENERIC=y
 CONFIG_PCI_XGENE=y
 CONFIG_PCIE_ALTERA=y
 CONFIG_PCIE_ALTERA_MSI=y
+CONFIG_PCI_THUNDER=y
 CONFIG_PCI_HOST_THUNDER_PEM=y
 CONFIG_PCI_HOST_THUNDER_ECAM=y
 CONFIG_PCIE_ROCKCHIP_HOST=m
diff --git a/drivers/pci/controller/Kconfig b/drivers/pci/controller/Kconfig
index adddf21fa381..28335ffa5d48 100644
--- a/drivers/pci/controller/Kconfig
+++ b/drivers/pci/controller/Kconfig
@@ -286,6 +286,13 @@ config PCI_LOONGSON
 	  Say Y here if you want to enable PCI controller support on
 	  Loongson systems.
 
+config PCI_THUNDER
+       bool "Thunder X PCIE controllers"
+       depends on ARM64
+       select PCI_QUIRKS
+       help
+          Say Y here to enable ThunderX ECAM and PEM PCI controllers.
+
 source "drivers/pci/controller/dwc/Kconfig"
 source "drivers/pci/controller/mobiveil/Kconfig"
 source "drivers/pci/controller/cadence/Kconfig"
diff --git a/drivers/pci/controller/Makefile b/drivers/pci/controller/Makefile
index efd9733ead26..8fad4781a5d3 100644
--- a/drivers/pci/controller/Makefile
+++ b/drivers/pci/controller/Makefile
@@ -45,8 +45,8 @@ obj-y				+= mobiveil/
 # ARM64 and use internal ifdefs to only build the pieces we need
 # depending on whether ACPI, the DT driver, or both are enabled.
 
+obj-$(CONFIG_PCI_THUNDER) += pci-thunder-ecam.o
+obj-$(CONFIG_PCI_THUNDER) += pci-thunder-pem.o
 ifdef CONFIG_PCI
-obj-$(CONFIG_ARM64) += pci-thunder-ecam.o
-obj-$(CONFIG_ARM64) += pci-thunder-pem.o
 obj-$(CONFIG_ARM64) += pci-xgene.o
 endif

From patchwork Tue Aug  4 12:44:16 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 11700371
Return-Path: <SRS0=wIAV=BO=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id BE71F722
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:47:19 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id B8818208C7
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:47:19 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=linaro.org header.i=@linaro.org
 header.b="nMnqkLp9"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726789AbgHDMqy (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Aug 2020 08:46:54 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:41168 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727997AbgHDMoW (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Aug 2020 08:44:22 -0400
Received: from mail-wm1-x342.google.com (mail-wm1-x342.google.com
 [IPv6:2a00:1450:4864:20::342])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 4D7EBC061757
        for <kvm@vger.kernel.org>; Tue,  4 Aug 2020 05:44:22 -0700 (PDT)
Received: by mail-wm1-x342.google.com with SMTP id f18so2127333wmc.0
        for <kvm@vger.kernel.org>; Tue, 04 Aug 2020 05:44:22 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=kh3fLdhtAK8iltYl6aGsCpfSwDzQcJQDII6+Lr3Ir4w=;
        b=nMnqkLp9MyaD7mVBWPOdpipOJAA/A+eqGB2+7v2AnGvajHcTmtJGzfEa8AObnk3OGO
         X6yzj+zqibdmVd52uW6BsVLE1twlWOD8iHOgBNZ4VGgtFVF3pWYXGAqPfb8l+53YHRNG
         fkEUw2haM1Xyld8Y5Y4NGy7cZi9KRdEu+s1yEngpns9vBqo+5u5vyUfrDXyt3ivwb/rB
         clg/7E3xKGUCx24uRr1RBdN9jkogBMFaY3KQRg6bWxxmOvo5fYHBnXt+k0Nf+/PDBu2f
         s6HexrWOx4jElhs7Dd34qf7UdiQd6LCDGnCq2XOnNawU9uylFjuZkOHE2fgbsJFmSbff
         SnFA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=kh3fLdhtAK8iltYl6aGsCpfSwDzQcJQDII6+Lr3Ir4w=;
        b=b2bwo30bVAE9CIBCoGFqIkXU3uuUN+UhO1eEyOnvxP7f4dcwXEum51MGmnfXHvykbY
         hq8d6OS59HiwQF/yUE6RNHsc8muWiX5AVx3VMpvi3fI97J43f0xhM+7nfyCMH/LEJQnm
         9NEH1WYKJ8HlyfWlJeDYWILDcGJ+pKXoZvxWdD9UuWUqoNrLvGqma3Vu5vETeaSM7tYR
         FQmQB6TtHph107xrLXYkXoc6bKa8lgZ8aWjFjLbhA+kQOipF3vuMfPfEca1gNBFnPwJ2
         m97xdWGL1cSuiFTrSP1jiBatfXl1jh3MQ9L62fHP5q8axozGcpIgZ84xufj5X50a6Cu7
         uUMg==
X-Gm-Message-State: AOAM530ftLk53awPgb66Bw4CRVkm35MH427ZjZvjSOy3bR4V8W/ZUJRR
        Zri/KpEZlTgZ3sfr4BUZTGf3Wg==
X-Google-Smtp-Source: 
 ABdhPJxlETqHM58ekeF7YRtlEWgL6bu7W0dZE0N/3gry/XAbgEfMhCjsPaytXate5VOSsOPSdbiQrg==
X-Received: by 2002:a1c:7702:: with SMTP id t2mr3859731wmi.169.1596545060810;
        Tue, 04 Aug 2020 05:44:20 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 r206sm4554096wma.6.2020.08.04.05.44.18
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 04 Aug 2020 05:44:18 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 9218B1FF8C;
        Tue,  4 Aug 2020 13:44:17 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org, linux-arm-kernel@lists.infradead.org
Cc: linux-kernel@vger.kernel.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [PATCH  v1 2/3] arm64: gate the whole of pci-xgene on
 CONFIG_PCI_XGENE
Date: Tue,  4 Aug 2020 13:44:16 +0100
Message-Id: <20200804124417.27102-3-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20200804124417.27102-1-alex.bennee@linaro.org>
References: <20200804124417.27102-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

This is a little weirder as bits of the file are already conditioned
on the exiting symbol. Either way they are not actually needed for
non-xgene machines saving another 12k:

-rwxr-xr-x 1 alex alex  86033880 Aug  3 16:39 vmlinux.orig*
-rwxr-xr-x 1 alex alex  85652472 Aug  3 16:54 vmlinux.rm-thunder*
-rwxr-xr-x 1 alex alex  85639808 Aug  3 17:12 vmlinux*

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
---
 drivers/pci/controller/Makefile | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/pci/controller/Makefile b/drivers/pci/controller/Makefile
index 8fad4781a5d3..3b9b72f5773a 100644
--- a/drivers/pci/controller/Makefile
+++ b/drivers/pci/controller/Makefile
@@ -47,6 +47,4 @@ obj-y				+= mobiveil/
 
 obj-$(CONFIG_PCI_THUNDER) += pci-thunder-ecam.o
 obj-$(CONFIG_PCI_THUNDER) += pci-thunder-pem.o
-ifdef CONFIG_PCI
-obj-$(CONFIG_ARM64) += pci-xgene.o
-endif
+obj-$(CONFIG_PCI_XGENE) += pci-xgene.o

From patchwork Tue Aug  4 12:44:17 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Patchwork-Submitter: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
X-Patchwork-Id: 11700363
Return-Path: <SRS0=wIAV=BO=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A98B9722
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:46:16 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id A1C9222C9F
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Aug 2020 12:46:16 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=linaro.org header.i=@linaro.org
 header.b="Riv8iutg"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727987AbgHDMoZ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Aug 2020 08:44:25 -0400
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:41158 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726210AbgHDMoV (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Aug 2020 08:44:21 -0400
Received: from mail-wr1-x441.google.com (mail-wr1-x441.google.com
 [IPv6:2a00:1450:4864:20::441])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id E6782C06174A
        for <kvm@vger.kernel.org>; Tue,  4 Aug 2020 05:44:20 -0700 (PDT)
Received: by mail-wr1-x441.google.com with SMTP id z18so33702416wrm.12
        for <kvm@vger.kernel.org>; Tue, 04 Aug 2020 05:44:20 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=linaro.org; s=google;
        h=from:to:cc:subject:date:message-id:in-reply-to:references
         :mime-version:content-transfer-encoding;
        bh=CYDox3KQ1JETO7R6nCajEi5q9XM0wuEKwlQrdfldtzE=;
        b=Riv8iutgfQUgNDMN92DtQvb/ueHJ6AIreMGAIsPCtm4ILvbZOn+PBNFru/JxFRQn+a
         9TfiKx4DTAO1/mtV9wsMqag4ynofuu+qJ8KcnT0H01tQepbNaqDTZvqQvMuZtPQBOduf
         CxLDVXhPdqFLT0zbGguTGrJKZfH+S/lGnAzhvNTBi0FcUKC5HJp/+r/qZDgMqCN1u61M
         ZIBRfcVMUR48Uz0nrWv/JeBjsaFxjG5I/w8jfhHReJuMa+Pm25oX2mloao7+N1Yr/e42
         uovU2IyFUipCeXc7bYv0WjUNANHbOH3ka4foDQ5we7li93TpmE8H4yyBhs21Q8mbmEQA
         00PQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=CYDox3KQ1JETO7R6nCajEi5q9XM0wuEKwlQrdfldtzE=;
        b=sZFEiDBrVbZJXjMpuuuw7OkGxU4lrmGi+gxJQ1/dFnX6/MoLgBtw/NBoaNj4DUAm4D
         7rvaSIuv9yOzwqzRMeOTxaF8vnNR0JL7BZYGN+Xa+skSkYhIgzaVelPHJ8v/ZVGw2XRd
         jUx/284n6UID6VFvbh3AzsVhyGRRhgkriYYAuhHFNyO7ZmwXcAMn/V3Vqd+hRJnallgb
         EChx6ReTx+f6bsLTTb1IuWKqUE6/6pnkkcUsCV1SrHng+ahfT20gy+xk06MOqMrDP86+
         YeSvvLpZnE3L2ogR+KsMTIrUyoWWWici2bAHxWiJm8dU39twMqicoNkEw09A2s+BijQr
         0TjA==
X-Gm-Message-State: AOAM533ylevJV7pRCvyph7K7Yu1UEuoT/5lLtlcwJQAY2eznOCxfR9Pl
        yAGUZHQzI43dnAQ0xJgIi4H8AA==
X-Google-Smtp-Source: 
 ABdhPJy369HvI22rb0UHPSZq9AsVjFavkg1pT+b1lsUC4Du192zTR5pC4Bv0+KzFRIkrLbZspUmmgg==
X-Received: by 2002:a5d:6a4a:: with SMTP id
 t10mr20525689wrw.360.1596545059657;
        Tue, 04 Aug 2020 05:44:19 -0700 (PDT)
Received: from zen.linaroharston ([51.148.130.216])
        by smtp.gmail.com with ESMTPSA id
 z207sm4576048wmc.2.2020.08.04.05.44.18
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 04 Aug 2020 05:44:18 -0700 (PDT)
Received: from zen.lan (localhost [127.0.0.1])
        by zen.linaroharston (Postfix) with ESMTP id 9DC9E1FF8F;
        Tue,  4 Aug 2020 13:44:17 +0100 (BST)
From: =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
To: kvm@vger.kernel.org, linux-arm-kernel@lists.infradead.org
Cc: linux-kernel@vger.kernel.org, kvmarm@lists.cs.columbia.edu,
 christoffer.dall@arm.com, maz@kernel.org,
 =?utf-8?q?Alex_Benn=C3=A9e?= <alex.bennee@linaro.org>
Subject: [PATCH  v1 3/3] kernel/configs: don't include PCI_QUIRKS in KVM guest
 configs
Date: Tue,  4 Aug 2020 13:44:17 +0100
Message-Id: <20200804124417.27102-4-alex.bennee@linaro.org>
X-Mailer: git-send-email 2.20.1
In-Reply-To: <20200804124417.27102-1-alex.bennee@linaro.org>
References: <20200804124417.27102-1-alex.bennee@linaro.org>
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The VIRTIO_PCI support is an idealised PCI bus, we don't need a bunch
of bloat for real world hardware for a VirtIO guest.

Signed-off-by: Alex Bennée <alex.bennee@linaro.org>
---
 kernel/configs/kvm_guest.config | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/configs/kvm_guest.config b/kernel/configs/kvm_guest.config
index 208481d91090..672863a2fdf1 100644
--- a/kernel/configs/kvm_guest.config
+++ b/kernel/configs/kvm_guest.config
@@ -13,6 +13,7 @@ CONFIG_IP_PNP_DHCP=y
 CONFIG_BINFMT_ELF=y
 CONFIG_PCI=y
 CONFIG_PCI_MSI=y
+CONFIG_PCI_QUIRKS=n
 CONFIG_DEBUG_KERNEL=y
 CONFIG_VIRTUALIZATION=y
 CONFIG_HYPERVISOR_GUEST=y
